runtime/js/effect.js

```diff
@@ -147,7 +201,7 @@ function caml_alloc_stack(hv, hx, hf) {
     // Call [hx] in the parent fiber
     return call(2, e);
   }
-  return [0, hval, [0, hexn, 0], [0, hv, hx, hf], 0];
+  return { k: hval, x: [hexn, 0], h: handlers, e: 0 };
```

Comment:
We can use an object rather than an array to implement the exception stack.
```suggestion
  return { k: hval, x: { h:hexn, t:0 }, h: handlers, e: 0 };
```